File: src/main/resources/static/js/app/EmployeeService.js
'use strict'

angular.module('demo.services', []).factory('EmployeeService',
    [ "$http", "CONSTANTS", function($http, CONSTANTS) {
        var service = {};
        service.getEmployeeById = function(userId) {
            return $http.get(CONSTANTS.getEmployeeByIdUrl + userId);
        }
        service.getPageEmployees = function() {
            return $http.get(CONSTANTS.getPageEmployees + "/1");
        }
        service.getListEmployees = function(empDto) {
            return $http.get(CONSTANTS.saveEmployee, empDto);
        }
        service.saveEmployee = function(empDto) {
            return $http.post(CONSTANTS.saveEmployee, empDto);
        }
        service.updateEmployee = function(empDto) {
            return $http.put(CONSTANTS.saveEmployee, empDto);
        }
        service.deleteEmployee = function(empDto) {
            return $http.delete(CONSTANTS.saveEmployee, empDto);
        }

        return service;
    } ]);

File: src/main/java/com/example/demo/service/EmployeeService.java
package com.example.demo.service;

import com.example.demo.dto.EmployeeDTO;
import com.example.demo.utils.Page;

import java.util.List;

public interface EmployeeService {

        EmployeeDTO getEmployeeById(Long empId);
        EmployeeDTO saveEmployee(EmployeeDTO empDto);

        List<EmployeeDTO> getAllEmployees();
        Page<EmployeeDTO> getPageEmployees(Integer pageNo, Integer size);

        void deleteEmployee(EmployeeDTO empDto);
        void deleteEmployeeById(Long empId);

        Page<EmployeeDTO> searchByName(Integer pageNo, Integer size, String query);

}
